core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -745,8 +745,9 @@ class DAGScheduler(
     submitWaitingStages()
   }
 
-  private[scheduler] def handleTaskSetFailed(taskSet: TaskSet, reason: String) {
-    stageIdToStage.get(taskSet.stageId).foreach {abortStage(_, reason) }
+  private[scheduler] def handleTaskSetFailed(taskSet: TaskSet, reason: String,
+                                             exception: Option[Throwable] = None) {
+    stageIdToStage.get(taskSet.stageId).foreach {abortStage(_, reason, exception) }
```

Comment:
nit:  multiline method defs should have each arg on a separate line, all indented 4 extra spaces.  Also include the return type (I know it wasn't there before but as long as you're touching this ...)

```scala
private[scheduler] def handleTaskSetFailed(
    taskSet: TaskSet,
    reason: String,
    exception: Option[Throwable] = None): Unit = {
```